collections/collections.go

```diff
@@ -75,3 +101,99 @@ func GetContentsInPath(coluuid string, path string, db *gorm.DB, u *util.User) (
 
 	return selectedRefs, nil
 }
+
+func GetDirectoryContents(c echo.Context, refs []util.ContentWithPath, queryDir, coluuid string) ([]CollectionListResponse, error) {
+	dirs := make(map[string]bool)
+	var result []CollectionListResponse
+	for _, r := range refs {
+		directoryContent, traversedDirectories, err := getDirectoryContent(r, dirs, queryDir, coluuid)
+		dirs = traversedDirectories
+
+		if err != nil {
+			if err == errors.New(ERR_INTERNAL_SERVER) {
```

Comment:
use util.HTTP error

Reply:
also, for 500, just return the error, the error handler will take care of the rest